bots/src/liquidator.rs

```diff
@@ -0,0 +1,418 @@
+        let mut current_offset = 0;
+        let mut params = json!({
+            "offset": current_offset,
+            "count": page_size,
+        });
+
+        while let Ok(page) = view::<BorrowPositions>(
+            &self.client,
+            self.market.clone(),
+            "list_borrow_positions",
+            params.clone(),
+        )
+        .await
+        {
+            let fetched = page.len();
+            all_positions.extend(page);
+            current_offset += page_size;
+            params["offset"] = current_offset.into();
+
+            if fetched < page_size {
+                break;
+            }
+        }
+
+        Ok(all_positions)
+    }
+
+    #[instrument(skip(self), level = "info")]
+    pub async fn run_liquidations(&self, concurrency: usize) -> anyhow::Result<()> {
+        let configuration = self.get_configuration().await?;
+        let oracle_response = self
+            .get_oracle_prices(
+                configuration.price_oracle_configuration.account_id.clone(),
+                &[
+                    configuration
+                        .price_oracle_configuration
+                        .borrow_asset_price_id,
+                    configuration
+                        .price_oracle_configuration
+                        .collateral_asset_price_id,
+                ],
+                configuration.price_oracle_configuration.price_maximum_age_s,
+            )
+            .await?;
+
+        futures::stream::iter(self.get_borrows().await?)
+            .map(|(borrow, position)| {
+                let oracle_response = oracle_response.clone();
+                let configuration = configuration.clone();
+                async move {
+                    self.try_liquidate(borrow, position, oracle_response, configuration)
+                        .await
+                }
+            })
+            .buffer_unordered(concurrency)
+            .collect::<Vec<_>>()
+            .await
+            .into_iter()
+            .collect::<anyhow::Result<Vec<_>>>()?;
+
+        Ok(())
+    }
+}
+
+#[instrument(level = "debug")]
+pub fn setup_liquidators(args: &Args) -> anyhow::Result<Vec<Liquidator<impl Swap>>> {
+    let client = JsonRpcClient::connect(args.network.get_rpc_url());
+    let signer =
+        InMemorySigner::from_secret_key(args.signer_account.clone(), args.signer_key.clone());
+    let asset = args.asset.clone();
+    let swap = match args.swap {
+        SwapType::RheaSwap => RheaSwap::new(
+            args.swap.get_account_id(args.network),
+            client.clone(),
+            signer.clone(),
+        ),
+    };
+
+    Ok(args
+        .markets
+        .iter()
+        .map(|market| {
+            Liquidator::new(
+                client.clone(),
+                signer.clone(),
+                asset.clone(),
+                market.clone(),
+                swap.clone(),
+                args.timeout,
+            )
+        })
+        .collect::<Vec<_>>())
```

Comment:
The explicit type annotation should be already implied to collect into a `Vec`. Simply `.collect()` should work, unless I am missing something.